### ISSUE TITLE
[resolved] CORS prefetch (`OPTIONS`) requests fail with a `400 Bad Request`

### ISSUE THREAD

**Issue by mbklein** [minio/minio#10002]
I am using a presigned URL to upload a file directly from the browser to S3, using minio via Docker in my dev and test environments. This has been working fine for months, but is broken as of `RELEASE.2020-07-02T00-15-09Z` and up through `latest`.

## Setup

I captured the `OPTIONS` request with the presigned URL from Chrome and tried to curl it. This URL won't work for testing; you'll need to generate a new presigned URL using whatever AWS S3 SDK you use.

```
curl 'http://localhost:9000/dev-uploads/ingest_sheets/582399ca-2f2b-425d-b408-d2f8f573fe3d.csv?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20200709%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20200709T184556Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=2b0a623c30d66208f4a2fccdcfb1946e82b88b5ff22a776b021166985b766c0b' \
  -X 'OPTIONS' \
  -H 'Access-Control-Request-Method: PUT' \
  -H 'Access-Control-Request-Headers: content-type' \
  -H 'Origin: http://localhost:3000' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-site' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: http://localhost:3000/project/c715812b-2e8e-45f8-ad14-0d18e73ac747/ingest-sheet/upload' \
  --compressed -v
```

## Expected Behavior

I would expect the communication between client and server to look like this:

```
> OPTIONS /dev-uploads/ingest_sheets/582399ca-2f2b-425d-b408-d2f8f573fe3d.csv?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20200709%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20200709T184556Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=2b0a623c30d66208f4a2fccdcfb1946e82b88b5ff22a776b021166985b766c0b HTTP/1.1
> Host: localhost:9000
> Access-Control-Request-Method: PUT
> Access-Control-Request-Headers: content-type
> Origin: http://localhost:3000
> Sec-Fetch-Mode: cors
> Sec-Fetch-Site: same-site
> Sec-Fetch-Dest: empty
> Referer: http://localhost:3000/project/c715812b-2e8e-45f8-ad14-0d18e73ac747/ingest-sheet/upload
>
< HTTP/1.1 200 OK
< Access-Control-Allow-Credentials: true
< Access-Control-Allow-Headers: Content-Type
< Access-Control-Allow-Methods: PUT
< Access-Control-Allow-Origin: http://localhost:3000
< Vary: Origin
< Vary: Access-Control-Request-Method
< Vary: Access-Control-Request-Headers
< Date: Thu, 09 Jul 2020 18:52:08 GMT
< Content-Length: 0
```

and that is in fact how it works prior to `RELEASE.2020-07-02T00-15-09Z`.

## Current Behavior

As of `RELEASE.2020-07-02T00-15-09Z`, the request returns an error:

```
> OPTIONS /dev-uploads/ingest_sheets/582399ca-2f2b-425d-b408-d2f8f573fe3d.csv?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20200709%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20200709T184556Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=2b0a623c30d66208f4a2fccdcfb1946e82b88b5ff22a776b021166985b766c0b HTTP/1.1
> Host: localhost:9000
> Access-Control-Request-Method: PUT
> Access-Control-Request-Headers: content-type
> Origin: http://localhost:3000
> Sec-Fetch-Mode: cors
> Sec-Fetch-Site: same-site
> Sec-Fetch-Dest: empty
> Referer: http://localhost:3000/project/c715812b-2e8e-45f8-ad14-0d18e73ac747/ingest-sheet/upload
>
< HTTP/1.1 400 Bad Request
< Accept-Ranges: bytes
< Content-Length: 363
< Content-Type: application/xml
< Server: MinIO/RELEASE.2020-07-02T00-15-09Z
< Date: Thu, 09 Jul 2020 18:51:39 GMT
<
<?xml version="1.0" encoding="UTF-8"?>
* Connection #0 to host localhost left intact
<Error><Code>XMinioUnknownAPIRequest</Code><Message>Unknown API request at /dev-uploads/ingest_sheets/582399ca-2f2b-425d-b408-d2f8f573fe3d.csv</Message><Resource>/dev-uploads/ingest_sheets/582399ca-2f2b-425d-b408-d2f8f573fe3d.csv</Resource><RequestId></RequestId><HostId>1a0e548d-173b-46cd-8ba3-36f1bec423d6</HostId></Error>
* Closing connection 0
```
## Steps to Reproduce

1. Generate a valid presigned URL to PUT a file into an existing bucket.
2. Use the above `curl` command with your presigned URL to simulate a CORS prefetch request

## Context

Users are unable to upload files from a browser directly to a presigned URL when using the latest build of minio. We are able to get our work done using `RELEASE.2020-06-22T03-12-50Z`, so it's not critical at this time.

## Your Environment

* Version used (`minio version`): `minio version RELEASE.2020-06-22T03-12-50Z`
* Environment name and version (e.g. nginx 1.9.1): `Docker Engine - Community` version `19.03.8`
* Server type and version: n/a
* Operating System and version (`uname -a`): n/a
* Link to your project: https://github.com/nulib/meadow

**Comment by harshavardhana** [comment]
Duplicate of #9979 , fixed in #9980

**Comment by mbklein** [comment]
Agh, sorry about that. I could have sworn I looked for a dupe. Thanks; I'll keep my eye out for a release.
